web-app/src/components/BookingCalendar/container.js

```diff
@@ -36,21 +40,65 @@ const BookingCalendarContainer = Wrapped =>
       this.props.setEventBeingUpdated(isBeingUpdated);
     };
 
-    onSelectSlot = ({ start, end }) => {
+    checkIfPastDatesSelected = start => {
       const today = new moment();
-      if (moment(start).isAfter(today.add(-1, 'days'))) {
+      return moment(start).isBefore(today);
+    };
+
+    selectedDatesOverlapExisting = (start, end) => {
+      const overlappingEvents = this.props.takenHolidays.filter(hol => {
+        if (hol.employee && hol.employee.employeeId === this.props.employeeId) {
+          var selectedDateRange = moment.range(
+            moment(start),
+            moment(end).endOf('day'),
+          );
+          var existingEvent = moment.range(moment(hol.start), moment(hol.end));
+          if (selectedDateRange.overlaps(existingEvent)) {
+            return true;
+          }
+        }
+      });
+      return overlappingEvents.length > 0 ? true : false;
```

Comment:
I don't think you need the `?  true : false` here. The comparison operator will return these anyway.